src/main/scala/com/cognite/sdk/scala/common/dataTypes.scala

```diff
@@ -12,7 +12,23 @@ import io.scalaland.chimney.dsl._
 final case class ItemsWithCursor[A](items: Seq[A], nextCursor: Option[String] = None)
 final case class Items[A](items: Seq[A])
 final case class ItemsWithIgnoreUnknownIds(items: Seq[CogniteId], ignoreUnknownIds: Boolean)
-final case class SdkException(message: String) extends Throwable(message)
+final case class SdkException(
+    message: String,
+    uri: Option[Uri] = None,
+    requestId: Option[String] = None
+) extends Throwable(message) {
+  override def toString: String = {
+    val uriMessage = uri.map(u => s", in response to request sent to ${u.toString()}").getOrElse("")
+    val requestIdMessage = requestId.map(id => s", with request id ${id}").getOrElse("")
+    val superString: String = super.toString
+    val superMessage = if (superString.length > 0) {
+      superString
+    } else {
+      "Missing error message"
+    }
+    s"$superMessage$uriMessage$requestIdMessage"
+  }
+}
```

Comment:
I thought this had been implemented already?

Reply:
If "this" refers to request ids, we have those in `CdpApiException`. That should be used for the JSON error responses from api.cognitedata.com, but we can also have errors that are not JSON errors, for example Bad Gateway, which is sent from Ambassador and doesn't have any body.
Those responses can however include a request id which might be useful.

Reply:
Ah, I see. Great :)